### PR TITLE
feat: add "type" field to JSON Schema, validation and asserts

### PR DESCRIPTION
**Pull request by digitalsadhu**
Begins work on https://github.com/eik-lib/issues/issues/1 by adding a `type` field to the schema

### CODE REVIEW

File: lib/schemas/eikjson.schema.json [hunk 1]
```diff
@@ -19,6 +19,12 @@
       "type": "string",
       "minLength": 1
     },
+    "type": {
+      "description": "The type of the Eik package. Must be one of 'package', 'npm' or 'map'",
```
Comment (gingermusketeer) [inline lib/schemas/eikjson.schema.json:23]: Wondering if this should help folks make a decision as to which one to choose. Would it be worth mentioning the matching url prefix? Does the different options need to be repeated from the enum below? Is this not included in the schema validation error?

**Review by gingermusketeer — APPROVED** [review]
Is it worth including a test for the error message to verify what consumers will experience if this field is incorrect?

File: lib/schemas/eikjson.schema.json (hunk 1 above)
Comment (digitalsadhu) [reply lib/schemas/eikjson.schema.json]: fixed

**Comment by digitalsadhu** [comment]
Asserted and improved the error message for type

**Review by gingermusketeer — APPROVED** [review]

**Comment by trygve-lie** [comment]
Great. Can you also update the readme before we merge this?

**Comment by digitalsadhu** [comment]
Done

**Pull request merged**